client/js/dnd.js

```diff
@@ -85,6 +73,22 @@ qq.DragAndDrop = function(o) {
         return parseEntryPromise;
     }
 
+    function extractDirectoryPath(entry) {
+        var name = entry.name,
+            fullPath = entry.fullPath,
+            indexOfNameInFullPath = fullPath.lastIndexOf(name);
+
+        // remove file name from full path string
+        fullPath = fullPath.substr(0, indexOfNameInFullPath);
+
+        // remove leading slash in full path string
+        if (fullPath.charAt(0) === "/") {
```

Comment:
This may be considered a breaking change, which I want to avoid at all costs. Was this required to fix the underlying issue?